### PR TITLE
fix(geomath): CartesianPolygon contains

### PR DESCRIPTION
**Pull request by paguos**
Signed-off-by: Pablo Osinaga <pablo.alberto.osinaga@fokus.fraunhofer.de>

## Type of this PR 

- [X] Bug fix
- [ ] Enhancement

## Description

Replace current implementation of the `contains` method in the `CartesianPolygon` class with a [PNPOLY algorithm](https://wrf.ecse.rpi.edu/Research/Short_Notes/pnpoly.html) implementation.

## Issue(s) related to this PR

Resolves internal issue 47.

## Affected parts of the online documentation

No.

## Definition of Done

- [X] You have read CONTRIBUTING.md carefully.
- [x] You have signed the [Contributor License Agreement](http://www.eclipse.org/legal/CLA.php).
- [X] All commits are signed-off (`-s`) with your mail address of your Eclipse Account.
- [X] `origin/main` has been merged into your Fork.
- [X] New functionality is covered by unit tests or integration tests. Code coverage must not decrease.
- [ ] There are no new SpotBugs warnings. 
- [x] Coding guidelines have been observed (see CONTRIBUTING.md).
- [x] All tests pass.

### CODE REVIEW

**Review by kschrab — CHANGES_REQUESTED** [review]
The contains calculation in GeoPolygon needs to be changed as well. You can put the PNPOLY algorithm to MathUtils or the like and reuse it in GeoPolygon

Thanks for the good test though! 👍

**Comment by paguos** [comment]
> The contains calculation in GeoPolygon needs to be changed as well. You can put the PNPOLY algorithm to MathUtils or the like and reuse it in GeoPolygon
> 
> Thanks for the good test though! +1

It was not working properly, I needed to be sure! :wink:

File: lib/mosaic-geomath/src/main/java/org/eclipse/mosaic/lib/geo/GeoPolygon.java [hunk 1]
```diff
@@ -80,25 +79,19 @@ private GeoRectangle calcBoundingBox(List<GeoPoint> vertices) {
...
-                intersections += intersect ? 1 : 0;
-            }
-            prev = curr;
+        float[] verticesXValues = new float[vertices.size()];
+        float[] verticesYValues = new float[vertices.size()];
+
+        for (int i = 0; i < vertices.size(); i++) {
+            GeoPoint geoPoint = vertices.get(i);
+            verticesXValues[i] = (float) geoPoint.toCartesian().getX();
+            verticesYValues[i] = (float) geoPoint.toCartesian().getY();
```
Comment (kschrab) [inline lib/mosaic-geomath/src/main/java/org/eclipse/mosaic/lib/geo/GeoPolygon.java:88]: I don't wanna be picky, but we could half the number of conversions to cartesian by converting to CartesianPoint once and use their X/Y coordinates afterwards.

Comment (paguos) [reply lib/mosaic-geomath/src/main/java/org/eclipse/mosaic/lib/geo/GeoPolygon.java:88]: You should be picky :+1:

**Review by kschrab — APPROVED** [review]
Thanks!

**Pull request merged**